src/cargo/core/compiler/output_sbom.rs

```diff
@@ -0,0 +1,245 @@
+//! cargo-sbom precursor files for external tools to create SBOM files from.
+//! See [`output_sbom`] for more.
+
+use std::collections::{BTreeMap, BTreeSet, HashMap};
+use std::path::PathBuf;
+
+use cargo_util_schemas::core::PackageIdSpec;
+use itertools::Itertools;
+use serde::Serialize;
+
+use crate::core::profiles::{DebugInfo, Lto, PanicStrategy, Profile};
+use crate::core::TargetKind;
+use crate::util::interning::InternedString;
+use crate::util::Rustc;
+use crate::CargoResult;
+
+use super::{BuildOutput, CompileMode};
+use super::{BuildRunner, Unit};
+
+/// Typed version of a SBOM format version number.
+#[derive(Serialize, Copy, Clone, Debug, Ord, PartialOrd, Eq, PartialEq)]
+pub struct SbomFormatVersion(u32);
+
+/// A profile can be overriden for individual packages.
+///
+/// This wraps a [`Profile`] object.
+/// See <https://doc.rust-lang.org/nightly/cargo/reference/profiles.html#overrides>
+#[derive(Serialize, Clone, Debug)]
+#[serde(rename_all = "snake_case")]
+struct SbomProfile {
+    name: InternedString,
+    opt_level: InternedString,
+    lto: Lto,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    codegen_backend: Option<InternedString>,
+    debuginfo: DebugInfo,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    split_debuginfo: Option<InternedString>,
+    debug_assertions: bool,
+    overflow_checks: bool,
+    rpath: bool,
+    panic: PanicStrategy,
+}
```

Comment:
Not a blocker but I would love to understand why some are not picked.

* codegen-units might have impact on final artifacts just like lto does.
* Why codegen-backend is selected, and rustflags isn't. Both of them are unstable though.

…and etc.

Reply:
For this initial implementation, I've dropped the `Profile` section. We can re-add it if needed when a rationale is available.